### PR TITLE
Add industry measures and clean industry pipeline

### LINKED ISSUE #41: Add proportion green tasks per SIC

### PR DESCRIPTION
**Pull request by lizgzil**
---

# Description

- Streamlines the industry measures in one place (keeps GreenMeasures a bit cleaner)
- Adds test
- Adds normalised emissions
- Adds green timeshares
- Fixes #41 #40 #21 
- Adds industry measures test

You can now call the industry measures like this:
```
from dap_prinz_green_jobs.pipeline.green_measures.industries.industries_measures_utils import IndustryMeasures
im = IndustryMeasures()
im.load_ch()
im.get_green_measure_for_company("Global British Petroleum")
>>> {'SIC': '09100', 'SIC_name': 'Support activities for petroleum and natural gas extraction', 'INDUSTRY TOTAL GHG EMISSIONS': 28.2, 'INDUSTRY GHG PER UNIT EMISSIONS': 0.05, 'INDUSTRY PROP HOURS GREEN TASKS': 17, 'INDUSTRY PROP WORKERS GREEN TASKS': 85.7, 'INDUSTRY PROP WORKERS 20PERC GREEN TASKS': 22.5}
im.get_green_measure_for_company("Boots")
>>> {'SIC': '46450', 'SIC_name': 'Wholesale of perfume and cosmetics', 'INDUSTRY TOTAL GHG EMISSIONS': 6299.6, 'INDUSTRY GHG PER UNIT EMISSIONS': 0.11, 'INDUSTRY PROP HOURS GREEN TASKS': 9.4, 'INDUSTRY PROP WORKERS GREEN TASKS': 39.7, 'INDUSTRY PROP WORKERS 20PERC GREEN TASKS': 17.4}
```

# Instructions for Reviewer

In order to test the code in this PR you need to ...

- run the above code
- run `pytest`
- run `python dap_prinz_green_jobs/pipeline/ojo_application/extract_green_measures.py` (could comment out non-industry bits tho to speed up)

Please pay special attention to ...

# Checklist:

- [ ] I have refactored my code out from `notebooks/`
- [ ] I have checked the code runs
- [ ] I have tested the code
- [ ] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [ ] I have documented the code
  - [ ] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [ ] I have explained this PR above
- [ ] I have requested a code review

### CODE REVIEW

File: dap_prinz_green_jobs/pipeline/green_measures/green_measures.py [hunk 1]
```diff
@@ -234,22 +236,11 @@ def get_industry_measures(self, job_advert: Dict[str, str]) -> List[dict]:
...
-            ind_green_measures_dict["INDUSTRY GHG EMISSIONS"] = [
-                im.get_green_industry_measure(
-                    company_name=comp_name,
-                    ghg_emissions_dict=self.ghg_emissions_dict,
-                    ojo_companies_house_dict=self.ojo_companies_house_dict,
-                )
-                for comp_name in comp_names
-            ]
-        else:
-            ind_green_measures_dict["INDUSTRY GHG EMISSIONS"] = None
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/green_measures.py:250]: nice - that is cleaner!

File: dap_prinz_green_jobs/pipeline/green_measures/industries/README.md [hunk 2]
```diff
@@ -20,5 +20,29 @@ contains functions needed to see whether industries are green or not.
...
+
+## Green measures
+
+Measures based of the job adverts predicted SIC:
+
+- INDUSTRY TOTAL GHG EMISSIONS : The 2020 total greenhouse gas emissions for this SIC section code (letter) or division code (two digit). From `atmosphericemissionsghg.xlsx`.
+- INDUSTRY GHG PER UNIT EMISSIONS : The 2021 greenhouse gas emissions per unit of economic activity for this SIC section code (letter) or division code (two digit). From `atmosphericemissionsghgintensity.xlsx`.
+- INDUSTRY PROP HOURS GREEN TASKS : The 2019 proportion of hours worked spent doing green tasks for this SIC section code (letter). From `greentasks.xlsx`.
+- INDUSTRY PROP WORKERS GREEN TASKS : The 2019 proportion of workers doing green tasks for this SIC section code (letter). From `greentasks.xlsx`.
+- INDUSTRY PROP WORKERS 20PERC GREEN TASKS : Thw 2019 proportion of workers spending at least 20% of their time doing green tasks for this SIC section code (letter). From `greentasks.xlsx`.
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/industries/README.md:36]: love!

File: dap_prinz_green_jobs/pipeline/green_measures/industries/README.md [hunk 3]
```diff
@@ -20,5 +20,29 @@ contains functions needed to see whether industries are green or not.
...
+
+## Usage
+
+```
+from dap_prinz_green_jobs.pipeline.green_measures.industries.industries_measures_utils import IndustryMeasures
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/industries/README.md:41]: I was able to run this in a notebook with no problems.

File: dap_prinz_green_jobs/pipeline/green_measures/industries/industries_measures_utils.py [hunk 4]
```diff
@@ -106,69 +122,280 @@ def get_ghg_sic(sic, ghg_emissions_dict: Dict[str, float] = ghg_emissions_dict):
...
+        )
+
+    def load_ch(self):
+        """
+        Keep this separate from init since it takes a while to load
+        """
+        # Dictionary of company names and companies house SIC data
+        self.ojo_companies_house_dict = load_companies_house_dict()
+
+    def get_ch_sic(
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/industries/industries_measures_utils.py:303]: note to change method names after we move away from using company house data

Comment (india-kerle) [reply dap_prinz_green_jobs/pipeline/green_measures/industries/industries_measures_utils.py:303]: + get rid of load_ch i'd guess

File: dap_prinz_green_jobs/tests/test_industry_measures.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
+import pytest
```
Comment (india-kerle) [inline dap_prinz_green_jobs/tests/test_industry_measures.py:1]: this passed for me when I ran pytest

**Review by india-kerle — APPROVED** [review]
thanks Liz - looks great - i love the new measures! I ran pytest and all tests passed + i tested instantiating industries measures and tried getting green measures for a given company and it also worked.

**Comment by lizgzil** [comment]
thanks for checking @india-kerle !

**Pull request merged**